pallets/xyk-liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,927 @@
+			let amm_pool_id = T::AMM::get_pair_id(asset_pair);
+			let yield_farm_id = T::LiquidityMiningHandler::stop_yield_farm(who.clone(), global_farm_id, amm_pool_id)?;
+
+			Self::deposit_event(Event::YieldFarmStopped {
+				global_farm_id,
+				yield_farm_id,
+				who,
+				asset_pair,
+			});
+
+			Ok(())
+		}
+
+		/// Resume yield farm for stopped yield farm.
+		///
+		/// This function resume incentivization from `GlobalFarm` and restore full functionality
+		/// for yield farm. Users will be able to deposit, claim and withdraw again.
+		///
+		/// WARN: Yield farm is NOT rewarded for time it was stopped.
+		///
+		/// Only farm owner can perform this action.
+		///
+		/// Parameters:
+		/// - `origin`: global farm's owner.
+		/// - `global_farm_id`: global farm id in which yield farm will be resumed.
+		/// - `yield_farm_id`: id of yield farm to be resumed.
+		/// - `asset_pair`: asset pair identifying yield farm in global farm.
+		/// - `multiplier`: yield farm multiplier in the farm.
+		///
+		/// Emits `YieldFarmResumed` event when successful.
+		#[pallet::weight(<T as Config>::WeightInfo::resume_yield_farm())]
+		pub fn resume_yield_farm(
+			origin: OriginFor<T>,
+			global_farm_id: GlobalFarmId,
+			yield_farm_id: YieldFarmId,
+			asset_pair: AssetPair,
+			multiplier: FarmMultiplier,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let amm_pool_id = Self::ensure_amm(asset_pair)?;
+
+			T::LiquidityMiningHandler::resume_yield_farm(
+				who.clone(),
+				global_farm_id,
+				yield_farm_id,
+				amm_pool_id,
+				multiplier,
+			)?;
+
+			Self::deposit_event(Event::<T>::YieldFarmResumed {
+				global_farm_id,
+				yield_farm_id,
+				who,
+				asset_pair,
+				multiplier,
+			});
+
+			Ok(())
+		}
+
+		/// Remove yield farm
+		///
+		/// This function marks a yield farm as ready to be removed from storage when it's empty. Users will
+		/// be able to only withdraw shares(without claiming rewards from yield farm). Unpaid rewards
+		/// will be transferred back to global farm and will be used to distribute to other yield farms.
+		///
+		/// Yield farm must be stopped before calling this function.
+		///
+		/// Only global farm's owner can perform this action. Yield farm stays in the storage until it's
+		/// empty(all farm entries are withdrawn). Last withdrawn from yield farm trigger removing from
+		/// the storage.
+		///
+		/// Parameters:
+		/// - `origin`: global farm's owner.
+		/// - `global_farm_id`: farm id from which yield farm should be destroyed.
+		/// - `yield_farm_id`: id of yield farm to be destroyed.
+		/// - `asset_pair`: asset pair identifying yield farm in the global farm.
+		///
+		/// Emits `YieldFarmDestroyed` event when successful.
+		#[pallet::weight(<T as Config>::WeightInfo::destroy_yield_farm())]
+		pub fn destroy_yield_farm(
+			origin: OriginFor<T>,
+			global_farm_id: GlobalFarmId,
+			yield_farm_id: YieldFarmId,
+			asset_pair: AssetPair,
+		) -> DispatchResultWithPostInfo {
```

Comment:
Why the return value is different here ? everywhere else is just DispatchResult.

I believe DispatchResult should work ?